public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tsx

```diff
@@ -76,6 +78,11 @@ export const Editor = (props: Props): JSX.Element => {
   const { datasource } = props;
   const { measurement, policy } = query;
 
+  useAsync(async () => {
+    const response = await getAllPolicies(datasource);
+    await setRetentionPolicies(response);
+  }, [datasource]);
```

Comment:
this can be even simpler, without needing `useState`, something like this:
```ts
  const policyData = useAsync(
        () => getAllPolicies(datasource),
        [datasource]
  );
  const retentionPolicies = policyData.error != null ? [] : policyData.value ?? [];
```